packages/core/src/External/External.ts

```diff
@@ -16,11 +16,13 @@ export class External {
   externalManager: ExternalManager;
   sourceFilePath: string;
   includedFiles: Set<string>;
+  userScriptsAndStyles: string[];
 
-  constructor(em: ExternalManager, srcFilePath: string) {
+  constructor(em: ExternalManager, srcFilePath: string, userScriptsAndStyles: string[]) {
     this.externalManager = em;
     this.sourceFilePath = srcFilePath;
     this.includedFiles = new Set([srcFilePath]);
+    this.userScriptsAndStyles = userScriptsAndStyles;
```

Comment:
Seems like code coverage has been reduced - can you add tests to cover "userScriptsAndStyles" ? especially as it seems you've changed the type.

Reply:
Noted, will work on boosting the test coverage. Thanks for the comment.

Reply:
Hi @kaixin-hc, after studying the code base,It seems that to write non-trivial test cases to cover the lines not covered (constructor for the external and external manager) is almost equivalent to writing test to the whole External package.

Currently, there is not test for the external package in the code base. So no available mocks to use.

While there are test for Site and Page, the existing mocks seems can not be modified to incorporate testing involving "External" without significant changes.

External manager is only used in "Site" as a parameter to call the "Generate" method of "Page". So I conclude that it impossible to test these lines non-trivially without creating the test for the External package as a whole, including related mocks and utils.

I wonder is my observation correct so far? Could I seek you help and enlighten me on how could I proceed from here? Thanks!!!

Reply:
Thanks for investigating! Yes, your observation seems correct. 

Tests for externals seem outside of the scope of this issue. For now, you can flag a separate issue to add tests for this and reduce the code coverage patch expected instead, so that the tests will pass.

Reply:
Could you help explaining how to temporarily remove/reduce the codecov/pathch requirement?

The latest PR that made change to external package this this "https://github.com/MarkBind/markbind/pull/2103", and it seems that there were not code coverage check then.

I tried to modify the CI.yaml file, but it seems does not work.

I try to search for potential config file in the repo, but could not find one.

The developer guide's fix bug section also did not cover tips regarding code coverage. 

I am stuck.

Reply:
Issue posted #2423.
Functional test added.

Reply:
Actually this is my bad. I saw this in a previous PR as well - your project coverage is the same even though the patch coverage dropped, and as you have observed this is because externals was not tested in the first place. So the changed lines in diff are flagged as not tested. 

No action is needed for this on your part; it passes CI so maintainers can merge it anyway. Sorry for the confusion!

Reply:
I see;
Noted, thank you for your comment!